File: src/main/java/com/drost/gradle/buildcache/minio/MinioPlugin.java
package com.drost.gradle.buildcache.minio;

import com.drost.gradle.buildcache.minio.internal.MinioBuildCacheServiceFactory;
import org.gradle.api.Plugin;
import org.gradle.api.initialization.Settings;
import org.gradle.caching.configuration.BuildCacheConfiguration;
import org.jetbrains.annotations.NotNull;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class MinioPlugin implements Plugin<Settings> {
    private static final Logger logger = LoggerFactory.getLogger(MinioPlugin.class);
    @Override
    public void apply(@NotNull Settings target) {
        logger.info("Apply Minio Build Cache to project");
        BuildCacheConfiguration buildCache = target.getBuildCache();
        buildCache.registerBuildCacheService(MinioBuildCache.class, MinioBuildCacheServiceFactory.class);
    }
}

File: src/main/kotlin/com/drost/gradle/buildcache/minio/MinioPlugin.kt
package com.drost.gradle.buildcache.minio

import com.drost.gradle.buildcache.minio.internal.MinioBuildCacheServiceFactory
import mu.KLogging
import org.gradle.api.Plugin
import org.gradle.api.initialization.Settings

class MinioPlugin : Plugin<Settings> {
    companion object : KLogging()

    override fun apply(settings: Settings) {
        logger.info { "apply cache plugin" }
        val buildCache = settings.buildCache
        buildCache.registerBuildCacheService(MinioBuildCache::class.java, MinioBuildCacheServiceFactory::class.java)
    }
}